File: game/projectilewars/scripts/vscripts/items/item_vision.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
require('items/base_item');
var item_rune_vision = /** @class */ (function (_super) {
    __extends(item_rune_vision, _super);
    function item_rune_vision() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return item_rune_vision;
}(item_base_rune));
LinkLuaModifier("modifier_rune_vision", "items/item_vision.lua", LuaModifierType.LUA_MODIFIER_MOTION_NONE);
var modifier_rune_vision = /** @class */ (function (_super) {
    __extends(modifier_rune_vision, _super);
    function modifier_rune_vision() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    modifier_rune_vision.prototype.DeclareFunctions = function () {
        return [
            modifierfunction.MODIFIER_PROPERTY_BONUS_DAY_VISION,
            modifierfunction.MODIFIER_PROPERTY_BONUS_NIGHT_VISION,
        ];
    };
    modifier_rune_vision.prototype.OnCreated = function () {
        this.vision = this.GetAbility().GetSpecialValueFor("bonus_vision");
    };
    modifier_rune_vision.prototype.GetBonusDayVision = function () {
        return this.vision;
    };
    modifier_rune_vision.prototype.GetBonusNightVision = function () {
        return this.vision;
    };
    return modifier_rune_vision;
}(CDOTA_Modifier_Lua));

File: game/projectilewars/scripts/vscripts/items/item_vision.lua
require("typescript_lualib")
require("items/base_item")
item_rune_vision = item_base_rune.new()
item_rune_vision.__index = item_rune_vision
item_rune_vision.__base = item_base_rune
function item_rune_vision.new(construct, ...)
    local instance = setmetatable({}, item_rune_vision)
    if construct and item_rune_vision.constructor then item_rune_vision.constructor(instance, ...) end
    return instance
end
LinkLuaModifier("modifier_rune_vision","items/item_vision.lua",LUA_MODIFIER_MOTION_NONE)
modifier_rune_vision = {}
modifier_rune_vision.__index = modifier_rune_vision
function modifier_rune_vision.new(construct, ...)
    local instance = setmetatable({}, modifier_rune_vision)
    if construct and modifier_rune_vision.constructor then modifier_rune_vision.constructor(instance, ...) end
    return instance
end
function modifier_rune_vision.DeclareFunctions(self)
    return {MODIFIER_PROPERTY_BONUS_DAY_VISION,MODIFIER_PROPERTY_BONUS_NIGHT_VISION}
end
function modifier_rune_vision.OnCreated(self)
    self.vision=CDOTABaseAbility.GetSpecialValueFor(CDOTA_Buff.GetAbility(self),"bonus_vision")
end
function modifier_rune_vision.GetBonusDayVision(self)
    return self.vision
end
function modifier_rune_vision.GetBonusNightVision(self)
    return self.vision
end
